integration-cli/docker_cli_run_test.go

```diff
@@ -2816,3 +2816,21 @@ func (s *DockerSuite) TestRunCapAddSYSTIME(c *check.C) {
 
 	dockerCmd(c, "run", "--cap-drop=ALL", "--cap-add=SYS_TIME", "busybox", "sh", "-c", "grep ^CapEff /proc/self/status | sed 's/^CapEff:\t//' | grep ^0000000002000000$")
 }
+
+// run create container failed should remove container
+func (s *DockerSuite) TestRunCreateContainerFaliedCleanUp(c *check.C) {
+	name := "flowers"
+	out, _, err := dockerCmdWithError("run", "--name", name, "--link", "nothing:nothing", "busybox")
+	if err == nil {
+		c.Fatal("Expected docker run to fail", out, err)
+	}
+
+	out, err = getAllContainers()
+	if err != nil {
+		c.Fatal(out, err)
+	}
+
+	if out != "" {
```

Comment:
c.Assert(out, check.Equals, "")

But also it may be better to instead just call `inspect